packages/phone_log/android/src/main/java/com/jiajiabingcheng/phonelog/PhoneLogPlugin.java

```diff
@@ -21,20 +28,71 @@
 /** PhoneLogPlugin */
 @TargetApi(Build.VERSION_CODES.LOLLIPOP)
 public class PhoneLogPlugin
-    implements MethodCallHandler, PluginRegistry.RequestPermissionsResultListener {
-  private final Registrar registrar;
+    implements MethodCallHandler, PluginRegistry.RequestPermissionsResultListener, FlutterPlugin, ActivityAware {
   private Result pendingResult;
+  private Registrar registrar;
+  // Activity for v2 embedding.
+  private Activity activity;
+  private MethodChannel methodChannel;
+  private Context context;
 
-  private PhoneLogPlugin(Registrar registrar) {
-    this.registrar = registrar;
+  public static void registerWith(Registrar registrar) {
+    PhoneLogPlugin instance = new PhoneLogPlugin();
+    instance.registrar = registrar;
+    instance.initInstance(registrar.messenger(), registrar.context());
+    registrar.addRequestPermissionsResultListener(instance);
   }
 
-  public static void registerWith(Registrar registrar) {
-    final MethodChannel channel =
-        new MethodChannel(registrar.messenger(), "github.com/jiajiabingcheng/phone_log");
-    PhoneLogPlugin phoneLogPlugin = new PhoneLogPlugin(registrar);
-    channel.setMethodCallHandler(phoneLogPlugin);
-    registrar.addRequestPermissionsResultListener(phoneLogPlugin);
+  private void initInstance(BinaryMessenger messenger, Context context) {
+    methodChannel =
+      new MethodChannel(messenger, "github.com/jiajiabingcheng/phone_log");
+    methodChannel.setMethodCallHandler(this);
+    this.context = context;
+  }
+
+  private Activity activity() {
+    return activity != null ? activity : registrar.activity();
+  }
+
+  @Override
+  public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
+    initInstance(binding.getBinaryMessenger(), binding.getApplicationContext());
+  }
+
+  @Override
+  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
+    methodChannel.setMethodCallHandler(null);
+    methodChannel = null;
+    context = null;
+  }
+
+  private void attachToActivity(ActivityPluginBinding activityPluginBinding) {
+    this.activity = activityPluginBinding.getActivity();
+    activityPluginBinding.addRequestPermissionsResultListener(this);
+  }
+
+  private void detachToActivity() {
+    this.activity = null;
```

Comment:
Keep the activityPluginBinding and removeRequestPermissionsResultListener when detaching.

Reply:
Ben? :)